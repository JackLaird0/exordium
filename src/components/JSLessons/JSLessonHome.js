import React, { Component } from 'react';

export class JSLessonHome extends Component {

  render() {
    return (
      <main>
        <header className="title-container">
          <h1 className="title">Welcome to Javascript!</h1>
        </header>
        <div className="body-container">
          <div className="content">
            <h1 className="overview">Overview</h1>
            <p className="text">
              JavaScript was created to make the web more dynamic. It is an object-oriented scripting language made to run inside a host environment like a web browser and provide programatic control over the objects of that environment. For example, when you click a button and you want something about the webpage to change, you will use JavaScript.
            </p>
            <h1 className="key-features">Key Features</h1>
            <p className="text">
              Understanding these four features will allow you to build dynamic applications with a high degree of interactivity
            </p>
            <li>Variables</li>
            <li>Functions</li>
            <li>Data Types</li>
            <li>Conditionals</li>
            <p className="text">
              Variables, functions, data types, and conditionals, are essential to understanding Javascript.  Understanding these core concepts allows for dynamic websites with a high level of user interactivity.
            </p>
          </div>
          <div className="instructions">
            <h1 className="lessons-games">Lessons & Games</h1>
            <p className="text">
              To get started, choose one of the lessons below. After you have completed the lesson, play the game to test your knowledge!
            </p>
            <div className="material">
              <div className="material-box">
                <h3 className="material-title">Lessons</h3>
                <div className="lesson-element">Variables</div>
                <div className="lesson-element">Declaring Functions</div>
                <div className="lesson-element">Loops</div>
                <div className="lesson-element">Data Types</div>
              </div>
              <div className="games">
                <h3 className="material-title">Games</h3>
                <div className="lesson-element">Variables</div>       
                <div className="lesson-element">Declaring Functions</div>
                <div className="lesson-element">Loops</div>
                <div className="lesson-element">Data Types</div>
              </div>
            </div>
          </div>
        </div>
      </main>
    );
  }
}

export default JSLessonHome;