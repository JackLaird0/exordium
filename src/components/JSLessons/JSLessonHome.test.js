import React from 'react';
import { shallow } from 'enzyme';
import JSLessonHome from './JSLessonHome';

describe('JSLessonHome', () => {

  it('should match the snapshot', () => {
    let wrapper = shallow(<JSLessonHome />)
    expect(wrapper).toMatchSnapshot();
  })
})