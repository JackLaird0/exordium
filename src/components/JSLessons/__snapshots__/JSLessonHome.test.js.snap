// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSLessonHome should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JSLessonHome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1>
                    Welcome to Javascript!
          </h1>
</header>,
        <div
          className="basics"
>
          <h1>
                    Before you start, please read this brief overview
          </h1>
          <p>
                    JavaScript allows us to manipulate our application using conditional logic
          </p>
          <p>
                    The key features of JavaScript include:
          </p>
          <li>
                    Variables
          </li>
          <li>
                    Functions
          </li>
          <li>
                    Data Types
          </li>
          <li>
                    Conditionals
          </li>
          <p>
                    Understanding these four features will allow you to build dynamic applications with a high degree of interactivity
          </p>
          <p>
                    Let's get started!
          </p>
</div>,
        <div
          className="instructions"
>
          <p>
                    To get started, choose one of the lessons below
          </p>
          <p>
                    After you have completed the lesson, play the game to test your knowledge!
          </p>
</div>,
        <div
          className="lessons"
>
          <div>
                    Lesson 01: Variables
          </div>
          <div>
                    Lesson 02: Declaring Functions
          </div>
          <div>
                    Lesson 03: Loops
          </div>
          <div>
                    Lesson 04: Data Types
          </div>
</div>,
        <div
          className="games"
>
          <div>
                    Game 01: Variables
          </div>
          <div>
                    Game 02: Declaring Functions
          </div>
          <div>
                    Game 03: Loops
          </div>
          <div>
                    Game 04: Data Types
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Welcome to Javascript!
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Javascript!",
          },
          "ref": null,
          "rendered": "Welcome to Javascript!",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Before you start, please read this brief overview
</h1>,
            <p>
              JavaScript allows us to manipulate our application using conditional logic
</p>,
            <p>
              The key features of JavaScript include:
</p>,
            <li>
              Variables
</li>,
            <li>
              Functions
</li>,
            <li>
              Data Types
</li>,
            <li>
              Conditionals
</li>,
            <p>
              Understanding these four features will allow you to build dynamic applications with a high degree of interactivity
</p>,
            <p>
              Let's get started!
</p>,
          ],
          "className": "basics",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Before you start, please read this brief overview",
            },
            "ref": null,
            "rendered": "Before you start, please read this brief overview",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "JavaScript allows us to manipulate our application using conditional logic",
            },
            "ref": null,
            "rendered": "JavaScript allows us to manipulate our application using conditional logic",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The key features of JavaScript include:",
            },
            "ref": null,
            "rendered": "The key features of JavaScript include:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Variables",
            },
            "ref": null,
            "rendered": "Variables",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Functions",
            },
            "ref": null,
            "rendered": "Functions",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Data Types",
            },
            "ref": null,
            "rendered": "Data Types",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Conditionals",
            },
            "ref": null,
            "rendered": "Conditionals",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Understanding these four features will allow you to build dynamic applications with a high degree of interactivity",
            },
            "ref": null,
            "rendered": "Understanding these four features will allow you to build dynamic applications with a high degree of interactivity",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Let's get started!",
            },
            "ref": null,
            "rendered": "Let's get started!",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              To get started, choose one of the lessons below
</p>,
            <p>
              After you have completed the lesson, play the game to test your knowledge!
</p>,
          ],
          "className": "instructions",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "To get started, choose one of the lessons below",
            },
            "ref": null,
            "rendered": "To get started, choose one of the lessons below",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "After you have completed the lesson, play the game to test your knowledge!",
            },
            "ref": null,
            "rendered": "After you have completed the lesson, play the game to test your knowledge!",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Lesson 01: Variables
</div>,
            <div>
              Lesson 02: Declaring Functions
</div>,
            <div>
              Lesson 03: Loops
</div>,
            <div>
              Lesson 04: Data Types
</div>,
          ],
          "className": "lessons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lesson 01: Variables",
            },
            "ref": null,
            "rendered": "Lesson 01: Variables",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lesson 02: Declaring Functions",
            },
            "ref": null,
            "rendered": "Lesson 02: Declaring Functions",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lesson 03: Loops",
            },
            "ref": null,
            "rendered": "Lesson 03: Loops",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lesson 04: Data Types",
            },
            "ref": null,
            "rendered": "Lesson 04: Data Types",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Game 01: Variables
</div>,
            <div>
              Game 02: Declaring Functions
</div>,
            <div>
              Game 03: Loops
</div>,
            <div>
              Game 04: Data Types
</div>,
          ],
          "className": "games",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Game 01: Variables",
            },
            "ref": null,
            "rendered": "Game 01: Variables",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Game 02: Declaring Functions",
            },
            "ref": null,
            "rendered": "Game 02: Declaring Functions",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Game 03: Loops",
            },
            "ref": null,
            "rendered": "Game 03: Loops",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Game 04: Data Types",
            },
            "ref": null,
            "rendered": "Game 04: Data Types",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
                        Welcome to Javascript!
            </h1>
</header>,
          <div
            className="basics"
>
            <h1>
                        Before you start, please read this brief overview
            </h1>
            <p>
                        JavaScript allows us to manipulate our application using conditional logic
            </p>
            <p>
                        The key features of JavaScript include:
            </p>
            <li>
                        Variables
            </li>
            <li>
                        Functions
            </li>
            <li>
                        Data Types
            </li>
            <li>
                        Conditionals
            </li>
            <p>
                        Understanding these four features will allow you to build dynamic applications with a high degree of interactivity
            </p>
            <p>
                        Let's get started!
            </p>
</div>,
          <div
            className="instructions"
>
            <p>
                        To get started, choose one of the lessons below
            </p>
            <p>
                        After you have completed the lesson, play the game to test your knowledge!
            </p>
</div>,
          <div
            className="lessons"
>
            <div>
                        Lesson 01: Variables
            </div>
            <div>
                        Lesson 02: Declaring Functions
            </div>
            <div>
                        Lesson 03: Loops
            </div>
            <div>
                        Lesson 04: Data Types
            </div>
</div>,
          <div
            className="games"
>
            <div>
                        Game 01: Variables
            </div>
            <div>
                        Game 02: Declaring Functions
            </div>
            <div>
                        Game 03: Loops
            </div>
            <div>
                        Game 04: Data Types
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Welcome to Javascript!
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Javascript!",
            },
            "ref": null,
            "rendered": "Welcome to Javascript!",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Before you start, please read this brief overview
</h1>,
              <p>
                JavaScript allows us to manipulate our application using conditional logic
</p>,
              <p>
                The key features of JavaScript include:
</p>,
              <li>
                Variables
</li>,
              <li>
                Functions
</li>,
              <li>
                Data Types
</li>,
              <li>
                Conditionals
</li>,
              <p>
                Understanding these four features will allow you to build dynamic applications with a high degree of interactivity
</p>,
              <p>
                Let's get started!
</p>,
            ],
            "className": "basics",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Before you start, please read this brief overview",
              },
              "ref": null,
              "rendered": "Before you start, please read this brief overview",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "JavaScript allows us to manipulate our application using conditional logic",
              },
              "ref": null,
              "rendered": "JavaScript allows us to manipulate our application using conditional logic",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "The key features of JavaScript include:",
              },
              "ref": null,
              "rendered": "The key features of JavaScript include:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Variables",
              },
              "ref": null,
              "rendered": "Variables",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Functions",
              },
              "ref": null,
              "rendered": "Functions",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Data Types",
              },
              "ref": null,
              "rendered": "Data Types",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Conditionals",
              },
              "ref": null,
              "rendered": "Conditionals",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Understanding these four features will allow you to build dynamic applications with a high degree of interactivity",
              },
              "ref": null,
              "rendered": "Understanding these four features will allow you to build dynamic applications with a high degree of interactivity",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Let's get started!",
              },
              "ref": null,
              "rendered": "Let's get started!",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                To get started, choose one of the lessons below
</p>,
              <p>
                After you have completed the lesson, play the game to test your knowledge!
</p>,
            ],
            "className": "instructions",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "To get started, choose one of the lessons below",
              },
              "ref": null,
              "rendered": "To get started, choose one of the lessons below",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "After you have completed the lesson, play the game to test your knowledge!",
              },
              "ref": null,
              "rendered": "After you have completed the lesson, play the game to test your knowledge!",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Lesson 01: Variables
</div>,
              <div>
                Lesson 02: Declaring Functions
</div>,
              <div>
                Lesson 03: Loops
</div>,
              <div>
                Lesson 04: Data Types
</div>,
            ],
            "className": "lessons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lesson 01: Variables",
              },
              "ref": null,
              "rendered": "Lesson 01: Variables",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lesson 02: Declaring Functions",
              },
              "ref": null,
              "rendered": "Lesson 02: Declaring Functions",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lesson 03: Loops",
              },
              "ref": null,
              "rendered": "Lesson 03: Loops",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lesson 04: Data Types",
              },
              "ref": null,
              "rendered": "Lesson 04: Data Types",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Game 01: Variables
</div>,
              <div>
                Game 02: Declaring Functions
</div>,
              <div>
                Game 03: Loops
</div>,
              <div>
                Game 04: Data Types
</div>,
            ],
            "className": "games",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Game 01: Variables",
              },
              "ref": null,
              "rendered": "Game 01: Variables",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Game 02: Declaring Functions",
              },
              "ref": null,
              "rendered": "Game 02: Declaring Functions",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Game 03: Loops",
              },
              "ref": null,
              "rendered": "Game 03: Loops",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Game 04: Data Types",
              },
              "ref": null,
              "rendered": "Game 04: Data Types",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
